### PR TITLE
Test refactor

### PR DESCRIPTION
**Pull request by nerfZael**
**Results of running tests**
Tests ran:
- `token/send/test_send_with_tasks.py::test_send_tokens_agent`
- `token/send/test_send_with_tasks.py::test_send_tokens_agent_with_check`
- `token/send/test_send_with_tasks.py::test_send_tokens_agent_with_check2`
- `token/send/test_send.py::test_auto_tx_send_eth`
- `token/send/test_send.py::test_auto_tx_send_erc20`
- `token/send/test_send.py::test_auto_tx_send_eth_sequential`
- `token/send/test_send.py::test_auto_tx_send_erc20_parallel`
- `token/test_swap.py::test_auto_tx_swap_eth`
- `token/test_swap.py::test_auto_tx_swap_multiple`
- `token/test_swap_and_send.py::test_auto_tx_swap_and_send_simple`
- `token/test_swap_and_send.py::test_auto_tx_swap_and_send_complex`

**Results:**
- `poetry run pytest -s ./autotx/tests/token/send/test_send_with_tasks.py::test_send_tokens_agent`
    Passes every time

- `poetry run pytest -s ./autotx/tests/token/send/test_send_with_tasks.py::test_send_tokens_agent_with_check`
    Sometimes fails
    Errors:
    - https://pastebin.com/9zmmJ2XA
    The LLM hallucinates the task result

- `poetry run pytest -s ./autotx/tests/token/send/test_send_with_tasks.py::test_send_tokens_agent_with_check2`
    Sometimes fails
    Errors: 
    - The LLM hallucinates the task result: https://pastebin.com/vxgM0J0f

- `poetry run pytest -s ./autotx/tests/token/send/test_send.py::test_auto_tx_send_eth`
    Sometimes fails
    Errors:
    - The define tasks step mentions gas fees, and then the agent sends 0.99 instead of 1 ETH to leave enough for gas fee even though it knows the user has 10 ETH:  https://pastebin.com/ZbfMw1nA

- `poetry run pytest -s ./autotx/tests/token/send/test_send.py::test_auto_tx_send_erc20`
    Sometimes fails
    Errors:
    - The LLM hallucinates the task result: https://pastebin.com/jwp7D8rM
    - The define tasks LLM uses agent that does not exist: https://pastebin.com/QDZU7ev8

- `poetry run pytest -s ./autotx/tests/token/send/test_send.py::test_auto_tx_send_eth_sequential`
    Sometimes fails
    Errors:
    - The LLM hallucinates the task result: https://pastebin.com/bVnfGX9d
    - The LLM hallucinates a function call instead of actual performing it: https://pastebin.com/Y21MKBDv

-  `poetry run pytest -s ./autotx/tests/token/send/test_send.py::test_auto_tx_send_erc20_parallel`
    Passes every time

- `poetry run pytest -s ./autotx/tests/token/test_swap.py::test_auto_tx_swap_eth`
    Passes every time

- `poetry run pytest -s ./autotx/tests/token/test_swap.py::test_auto_tx_swap_multiple`
    Fails every time
    Errors:
    - gnosis.safe.exceptions.CannotEstimateGas: Cannot estimate gas with `eth_estimateGas`: execution reverted: revert: STF: https://pastebin.com/6hFbHieV
    - Define tasks LLM thinks tasks are executed one by one (it needs to be aware of batching): https://pastebin.com/uibHuL9d

- `poetry run pytest -s ./autotx/tests/token/test_swap_and_send.py::test_auto_tx_swap_and_send_simple`
    Fails every time
    Errors:
    - Define tasks LLM thinks tasks are executed one by one so it check balances after tx: https://pastebin.com/CjAaMNPM

- `poetry run pytest -s ./autotx/tests/token/test_swap_and_send.py::test_auto_tx_swap_and_send_complex`
    Fails every time
    Errors:
    - Define tasks LLM thinks tasks are executed one by one so it check balances after tx: https://pastebin.com/9EN4rvVN

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,15 +7,15 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = ">=3.10,<3.13"
-crewai = "0.19.0"
 crewai-tools = "0.0.15"
```
Comment (cbrzn) [inline pyproject.toml:10]: do we want to also update `crewai-tools` if `crewai` was updated?

File: autotx/tests/token/test_swap_and_send.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+def test_auto_tx_swap_and_send_simple(configuration, auto_tx):
+    (_, _, client, manager) = configuration
+    web3 = load_w3()
+    network_info = SUPPORTED_NETWORKS.get(web3.eth.chain_id)
+    wbtc_address = ETHAddress(network_info.tokens["wbtc"], web3)
+
+    receiver = ETHAddress("0x10f8Bf6a479F320ead074411A4b0e7944eA8C9c1", client.w3)
+
+    prompt = f"Swap ETH to 0.05 WBTC, then, and send 0.01 WBTC to {receiver}"
```
Comment (cbrzn) [inline autotx/tests/token/test_swap_and_send.py:16]:
```suggestion
    prompt = f"Swap ETH to 0.05 WBTC, and send 0.01 WBTC to {receiver}"
```

File: autotx/tests/token/test_swap.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    assert balance + 100 == new_balance
+
+def test_auto_tx_swap_multiple(configuration, auto_tx):
+    (_, _, _, manager) = configuration
+    web3 = load_w3()
+    network_info = SUPPORTED_NETWORKS.get(web3.eth.chain_id)
+    usdc_address = ETHAddress(network_info.tokens["usdc"], web3)
+    wbtc_address = ETHAddress(network_info.tokens["wbtc"], web3)
+
+    prompt = "Buy 1000 USDC with ETH and then buy 0.05 WBTC with the USDC"
```
Comment (cbrzn) [inline autotx/tests/token/test_swap.py:30]: right now 1000 USDC ~= 0.014 WBTC - So this is not possible

File: autotx/tests/token/test_swap.py
Comment (cbrzn) [inline autotx/tests/token/test_swap.py:1]: nit: should we move `test_swap` to `swap` folder? like you did with `send`

Comment (nerfZael) [reply autotx/tests/token/test_swap.py:1]: We can, but I didn't feel the need to since it's the only thing that would be in the dir

**Review by cbrzn — APPROVED** [review]

**Pull request merged**